src/main/java/frc/robot/subsystems/NoteVision.java

```diff
@@ -111,6 +125,24 @@ public List<Pose2d> getNotes(Pose2d pose) {
         return poses;
     }
 
+    public boolean checkForNote(DriveTrain driveTrain, int wantedNote, NoteVision noteVision) {
+        
+        List<Pose2d> notes = noteVision.getNotes(driveTrain.getPose());
```

Comment:
This is the "cleanest" way to do this, but it would be more efficient to suck in the code from "getNotes(Pose2d)" here. Then you will only do the work that is absolutely needed. But it will mean duplicate code. Kind of hard call.

Reply:
what would make it more efficent

Reply:
If the loop from getNotes(Pose2d) was duplicated in this routine, you would save on:
* skip any entries as soon as you see a match
* save building the output List<Pose2d>
* not have to build the Pose2d (just a Translate2d to compute distance)

It is going to be small for us, so probably only worth it if we could get rid of getNotes(Pose2d) completely.

Reply:
i did it and kept the other one just in case